json-schema/json-schema/src/main/scala/endpoints4s/algebra/JsonSchemas.scala

```diff
@@ -349,6 +349,23 @@ trait JsonSchemas extends TuplesSchemas with PartialInvariantFunctorSyntax {
       tpe: JsonSchema[A]
   ): Record[Option[A]]
 
+  /** The JSON schema of a record with a single field `name` of type `A` with fine control
+    *  over presence and null
+    *
+    *   - Encoder interpreters may produce a JSON object with one property of the given `name`,
+    *     which can be set to the `null` value,
+    *   - Decoder interpreters successfully decode `Absent` if the field is absent, `Null` if
+    *     it is present but has the value `null`. They fail if the field is
+    *     present but contains an invalid value,
+    *   - Documentation interpreters produce the JSON schema of a JSON object with an
+    *     optional property of the given `name`.
+    *
+    * @group operations
+    */
+  def preciseField[A](name: String, documentation: Option[String] = None)(implicit
+      tpe: JsonSchema[A]
+  ): Record[JsonSchemas.PreciseField[A]]
```

Comment:
To “fix” the CI failure, we need to implement this method. Indeed, leaving it abstract in binary incompatible change (existing code that mixes-in the trait `JsonSchemas` won’t have an implementation of the method). In endpoints4s, we decided to handle such situations by providing a default implementation that throws an error at run-time. This is not a problem because existing code couldn’t call this method, so there is not much risk for them to get a run-time exception. The only issue that can happen is if someone wants to upgrade to the new version of the algebra without upgrading the interpreters: in that case, they will get the run-time exception. Therefore, the message in the exception should tell the users to upgrade their interpreters:

~~~ scala
sys.error("Unsupported algebra version: 1.11.0. Please update your interpreter dependency.")
~~~

(Note that we do have a method specifically for that purpose in the `algebra` module, but we can not use it from the `json-schema` module…)

Reply:
This is a nice trick ! It will be quite useful to do this in my $WORK code base.